slackbot_using_azure_openai/service.bal

```diff
@@ -0,0 +1,78 @@
+import ballerina/crypto;
+import ballerina/http;
+import ballerinax/azure.openai.chat;
+
+configurable string openAIToken = ?;
+configurable string serviceUrl = ?;
+configurable string deploymentId = ?;
+configurable string slackSigningSecret = ?;
+
+const API_VERSION = "2023-03-15-preview";
+const MAX_MESSAGES = 25;
+
+final chat:Client azureOpenAI = check new (
+    config = {auth: {apiKey: openAIToken}},
+    serviceUrl = serviceUrl
+);
+
+type ChatMessage record {|
+    string role;
+    string content;
+|};
+
+service /slack on new http:Listener(8080) {
+    map<ChatMessage[]> chatHistory = {};
+
+    resource function post events(http:Request request) returns json|error {
```

Comment:
Can we define record type for return type instead of json?

Reply:
Addressed in [6737695](https://github.com/ballerina-guides/ai-samples/pull/64/commits/6737695630388bbd0350796f53bcd230e7b02993)